### DISCUSSION TITLE
Q (plugin development): `ya.manager_emit` doesn't seem to do anything

### DISCUSSION THREAD

**Discussion by hankertrix** [sxyazi/yazi#1230] [1. Q&A] [answered]
I'm trying to make a plugin, but for some reason `ya.manager_emit` doesn't seem to perform any commands, despite the branch being hit.

Here's the source code of the plugin:
https://github.com/hankertrix/augment-command.yazi/blob/main/init.lua

An example is using the command to open the editor, the line https://github.com/hankertrix/augment-command.yazi/blob/main/init.lua#L341 is reached and the line is added to the log file, but the editor doesn't open.

Running the plugin with `plugin --sync` instead of `plugin` still results in the same problem, which is quite unexpected.

It probably has something to do with the async system, but I'm unsure where I'm going wrong. Please help.

**Comment by sxyazi** [comment, marked as answer]
The first argument of `ya.manager_emit()` is the command name without any parameters, while the parameters are in its second table. So you need:

```diff
- ya.manager_emit('shell "' .. command .. ' $0"', args)
+ ya.manager_emit('shell', { command .. ' "$0"', block = true, confirm = true })
```

**Reply by hankertrix** [reply]
This was exactly it! Thank you!
